### PR TITLE
feat: Add session secret env var check

### PR DESCRIPTION
**Pull request by Planlos5000**
Add documentation for the <code>SESSION_SECRET</code> env var.
Also add a check if the variable is set if an authentication provider is configured.
Add <code>SESSION_SECRET</code> to the docker compose.

## Description

Add the documentaion for the <code>SESSION_SECRET</code> environment variable.
Also add a check if the <code>SESSION_SECRET</code> is set if an authentication provider is set.

## Changelog

Add documentation for the <code>SESSION_SECRET</code> env var.
Add check if the variable is set if an authentication provider is configured.
Add <code>SESSION_SECRET</code> to the docker compose.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The light- and dark-theme are both supported and tested
- [x] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

### CODE REVIEW

File: server/src/main.go [hunk 1]
```diff
@@ -341,6 +347,10 @@ func run(c *cli.Context) error {
 		}
 	}
 
+	if c.String("session-secret") == "" && len(providersMap) != 0 {
+		return errors.New("you may not start the application without a session secret if an authentication provider is configured")
+	}
+
```
Comment (Schwehn42) [inline server/src/main.go:353]: nice little detail to also error check this 👍

Comment (Schwehn42) [inline server/src/main.go:353]: nice little detail to also error check this 👍

File: deployment/docker/docker-compose.yml [hunk 2]
```diff
@@ -30,6 +30,7 @@ services:
       # SCRUMLR_AUTH_OIDC_CLIENT_ID: "${OIDC_CLIENT_ID}"
       # SCRUMLR_AUTH_OIDC_CLIENT_SECRET: "${OIDC_CLIENT_SECRET}"
       # SCRUMLR_AUTH_OIDC_DISCOVERY_URL: "${OIDC_DISCOVERY_URL}"
+      # SESSION_SECRET: "${SESSION_SECRET}"
```
[the remark this answers is not in the document]
Comment (Planlos5000) [reply deployment/docker/docker-compose.yml:33]: This env var is used by gothic and must be named <code>SESSION_SECRET</code>

[the remark this answers is not in the document]
Comment (Schwehn42) [reply deployment/docker/docker-compose.yml:33]: ah I see 👍

**Review by Schwehn42 — APPROVED** [review]
lgtm 🙏

**Pull request merged**